### PR TITLE
Adjust manage languages (task #806)

### PR DESCRIPTION
**Pull request by stepanov**

### CODE REVIEW

File: src/Controller/TranslationsController.php [hunk 1]
```diff
@@ -94,6 +107,7 @@ public function addOrUpdate()
             throw new \RuntimeException('Wrong type of request!');
         }
         $params = $this->request->getData();
+        error_log(__METHOD__ . ": " . print_r($params, true), 3, '/tmp/qobo.log');
```
Comment (mamchenkov) [inline src/Controller/TranslationsController.php]: @stepanov please remove the debug.

File: src/Controller/TranslationsController.php [hunk 2]
```diff
@@ -108,8 +122,9 @@ public function addOrUpdate()
         }
 
         $translation = $this->Translations->patchEntity($translation, $params);
+        error_log(__METHOD__ . ": translation=" . print_r($translation, true), 3, '/tmp/qobo.log');
```
Comment (mamchenkov) [inline src/Controller/TranslationsController.php]: @stepanov please remove the debug.

File: src/Controller/TranslationsController.php [hunk 3]
```diff
@@ -108,8 +122,9 @@ public function addOrUpdate()
...
         $result = $this->Translations->save($translation);
-
+        error_log(__METHOD__ . ": result=" . print_r($result, true), 3, '/tmp/qobo.log');
```
Comment (mamchenkov) [inline src/Controller/TranslationsController.php]: @stepanov please remove the debug.

File: src/Template/Languages/add.ctp [hunk 4]
```diff
@@ -29,21 +29,22 @@ echo $this->Html->scriptBlock(
...
+                                    'type' => 'select',
+                                    'class' => 'select2',
+                                    'empty' => true
+                                ]);
+                            ?>
                         </div>
-                        <div class="col-md-6">
-                            <?= $this->Form->input('short_code'); ?>
+                        <div class="col-xs-6">
+                            <?= $this->Form->input('description'); ?>
```
Comment (mamchenkov) [inline src/Template/Languages/add.ctp]: @stepanov do we need to add the description of the language?  I think the language label should be sufficient.

File: src/Template/Languages/add.ctp [hunk 5]
```diff
@@ -29,21 +29,22 @@ echo $this->Html->scriptBlock(
...
                     </div>
                     <div class="row">
                         <div class="col-xs-12">
-                            <?= $this->Form->input('description'); ?>    
-                        </div>
-                    </div>
-                    <div class="row">
-                        <div class="col-xs-12">
-                            <?= $this->Form->input('is_active'); ?> 
+                            <?= $this->Form->input('is_rtl', ['label' => __('Right to left direction')]); ?>
```
Comment (mamchenkov) [inline src/Template/Languages/add.ctp]: @stepanov this shouldn't be left to the user.  If we know the languages, we know which ones are RTL and which ones are LTR.

File: config/translations.php [hunk 6]
```diff
@@ -0,0 +1,480 @@
+<?php 
+
+return [
+    'Translations' => [
+        'languages' => [
+            '' => '',
```
Comment (mamchenkov) [inline config/translations.php]: @stepanov do we allow for an empty language? I think we shouldn't.

File: config/translations.php [hunk 7]
```diff
@@ -0,0 +1,480 @@
...
+            'nl_AW' => 'Dutch (Aruba)',
+            'nl_BE' => 'Dutch (Belgium)',
+            'nl_BQ' => 'Dutch (Caribbean Netherlands)',
+            'nl_CW' => 'Dutch (Curaçao)',
+            'nl_NL' => 'Dutch (Netherlands)',
+            'nl_SX' => 'Dutch (Sint Maarten)',
+            'nl_SR' => 'Dutch (Suriname)',
+            'dz' => 'Dzongkha',
+            'dz_BT' => 'Dzongkha (Bhutan)',
+            'eo' => 'Esperanto',
```
Comment (mamchenkov) [inline config/translations.php]: @stepanov I think we are missing the English languages (all except the 'en') ... like the American English, etc.

**Comment by stepanov** [comment]
Requested changes are implemented:
1. Link between languages and translations is still done via ID but language is just marked as deleted using  Muffin/Trash plugin and restored instead of adding as a new.
2. Edit language is disabled.
3. View language is disabled.
4. Already added and non-deleted languages are removed from the list when new language is added.

File: src/Controller/LanguagesController.php [hunk 8]
```diff
@@ -50,15 +62,27 @@ public function add()
     {
         $language = $this->Languages->newEntity();
         if ($this->request->is('post')) {
-            $language = $this->Languages->patchEntity($language, $this->request->getData());
+            $data = $this->request->getData();
+            $data['is_rtl'] = $this->_setDirection($data);
+            $data['name'] = $this->Language->languages[$data['code']];
```
Comment (georgeconstantinou) [inline src/Controller/LanguagesController.php]: @stepanov I think this is a leftover, as from what I understand we don't use language name anymore.

Comment (stepanov) [inline src/Controller/LanguagesController.php]: @georgeconstantinou just pushed updated migration file

**Pull request merged**